apax/md/simulate.py

```diff
@@ -285,7 +297,19 @@ def body_fn(i, state):
         leave=True,
     )
     while step < n_outer:
-        new_state, neighbor, current_temperature = sim(state, step, neighbor)
+        new_state, neighbor, current_temperature, all_checks_passed = sim(state, step, neighbor)
+
+        if np.any(np.isnan(state.position)) or np.any(np.isnan(state.velocity)):
+            raise ValueError(
+                f"NaN encountered, simulation aborted after {step+1} steps."
+            )
+
+        if not all_checks_passed:
+            with logging_redirect_tqdm():
+                log.info(
+                    f"One or more dynamics checks failed at step: {step+1}"
+                )
```

Comment:
A failed check is not `log.info` but more `log.warning` or even `log.critical` or `log.error`?

Reply:
Yeah fair. in IPS it is log.critical, I'll change it to that.